Can this library run in WASM?

I'm looking to deploy my code to Cloudfare workers, and it uses this library. I was wondering, do you plan to support WASM as build target?

I just tried now, and had a small error, due to the `hostname` dependency (maybe others?).

```
error: Unsupported target OS! Create an issue: https://github.com/svartalf/hostname/issues/new
  --> /Users/amaurymartiny/.cargo/registry/src/github.com-1ecc6299db9ec823/hostname-0.3.1/src/lib.rs:81:9
   |
81 |         compile_error!("Unsupported target OS! Create an issue: https://github.com/svartalf/hostname/issues/new");
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0433]: failed to resolve: use of undeclared type or module `sys`
   --> /Users/amaurymartiny/.cargo/registry/src/github.com-1ecc6299db9ec823/hostname-0.3.1/src/lib.rs:105:5
    |
105 |     sys::get()
    |     ^^^ use of undeclared type or module `sys`

error: aborting due to 2 previous errors
```

We could return `localhost` in wasm.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -43,6 +43,16 @@ jobs:
         command: test
         args: --all --no-default-features --features runtime-async-std
 
+  # Check WASI 0.2 target.
+  check_wasi:
+    name: Check wasm32-wasip2 target
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - run: rustup target add wasm32-wasip2
+      - name: check
+        run: cargo check --target wasm32-wasip2
+
   check_fmt_and_docs:
     name: Checking fmt and docs
     runs-on: ubuntu-latest

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,7 +9,7 @@ license = "MIT OR Apache-2.0"
 authors = ["dignifiedquire <me@dignifiedquire.com>"]
 categories = ["email"]
 keywords = ["email", "smtp", "mailer"]
-edition = "2018"
+edition = "2021"
 
 [badges]
 maintenance = { status = "actively-developed" }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,7 +25,7 @@ futures = "0.3.21"
 log = "^0.4"
 nom = "^7.0"
 thiserror = "1"
-tokio = { version = "1", features = ["time", "io-util"], optional = true }
+tokio = { version = "1", default-features = false, features = ["time", "io-util"], optional = true }
 
 [dev-dependencies]
 env_logger = "^0.9"

